analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpArgumentTracker.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace SonarAnalyzer.Trackers
+{
+    internal class CSharpArgumentTracker : ArgumentTracker<SyntaxKind>
+    {
+        protected override SyntaxKind[] TrackedSyntaxKinds => new[]
+            {
+                SyntaxKind.AttributeArgument,
+                SyntaxKind.Argument,
+            };
+
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+        protected override IReadOnlyCollection<SyntaxNode> ArgumentList(SyntaxNode argumentNode) =>
+            argumentNode switch
+            {
+                AttributeArgumentSyntax { Parent: AttributeArgumentListSyntax { Arguments: { } list } } => list,
+                ArgumentSyntax { Parent: ArgumentListSyntax { Arguments: { } list } } => list,
+                ArgumentSyntax { Parent: BracketedArgumentListSyntax { Arguments: { } list } } => list,
```

Comment:
You can replace these two with one line where Parent is BaseArgumentListSyntax. The Arguments property seems to be on the base class.